ember-server/shared/src/main/scala/org/http4s/ember/server/internal/ServerHelpers.scala

```diff
@@ -340,7 +340,9 @@ private[server] object ServerHelpers extends ServerHelpersPlatform {
               .withLogger(tlsLogger)
               .build
               .evalMap(tlsSocket =>
-                tlsSocket.write(fs2.Chunk.empty) >>
+                tlsSocket
+                  .write(fs2.Chunk.empty)
+                  .handleErrorWith(err => logger.warn(err)("Failed to upgrade socket to TLS")) >>
```

Comment:
This completely suppresses the error—we no longer have short-circuiting. That means that `upgradeSocket` will return normally and return a dud socket.
https://github.com/http4s/http4s/blob/00165289e755e9c5d4b13451db3e33da05512a23/ember-server/shared/src/main/scala/org/http4s/ember/server/internal/ServerHelpers.scala#L195-L198

Reply:
One possible solution would be to do something like `handleErrorWith(e => Async[F].raiseError(CustomErrorType(e)))`

And pattern match it out here:
https://github.com/http4s/http4s/blob/00165289e755e9c5d4b13451db3e33da05512a23/ember-server/shared/src/main/scala/org/http4s/ember/server/internal/ServerHelpers.scala#L288C1-L292C12

Reply:
Can we do something similar to this PR?
- https://github.com/http4s/http4s/pull/7157

Reply:
That could work, I think. I will have a deeper look tomorrow :)